browser/main/lib/dataApi/attachmentManagement.js

```diff
@@ -38,26 +114,34 @@ function copyAttachment (sourceFilePath, storageKey, noteKey, useRandomName = tr
     }
 
     try {
-      if (!fs.existsSync(sourceFilePath)) {
-        reject('source file does not exist')
+      const isBase64 = typeof sourceFilePath === 'object' && sourceFilePath.type === 'base64'
+      if (!fs.existsSync(sourceFilePath) && !isBase64) {
```

Comment:
Why do you reject images that are not base64? e.g. i might drag&drop a pdf. Are all files always base64?!

Reply:
Only if the file not exists and the file is not a base64, reject.
In additional, I think you should add `return`  before `reject`, the code should not run through if the file not exists.

Reply:
yes, but i mean what happens if you drop a PDF for example. Is that still possible with your changes? Or will it be rejected because it is no pdf?

Reply:
This patch does nothing for other file. It will work same as before

Reply:
It has many dependencies, maybe we don't need to import so many code.

Reply:
I agree with @ehhc. This feature looks good to have. But I think it should not be included in the code base of boostnote. It is almost impossible to maintain...

Reply:
Why don't we make a npm module for this? And, I think we need some test for this too.

Reply:
Ok, I'll do it some days later.